examples/singletons.rb

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env ruby
+example_description = <<DESC
+
+  Sub Plans Example
+  ===================
+
+  This example shows, how singleton actions can be used for making sure
+  there is only one instance of the action running at a time.
+
+  Singleton actions try to obtain a lock at the beggining of their plan
+  phase and fail if they can't do so. In run phase they check if they
+  have the lock and try to acquire it again if they don't. These actions
+  release the lock at the end of their finalize phase.
+
+DESC
+
+require_relative 'example_helper'
+
+class SingletonExample < Dynflow::Action
+  middleware.use ::Dynflow::Middleware::Common::Singleton
+  include Dynflow::Action::Singleton
```

Comment:
what about putting this into the `Dynflow::Actions::Singleton`:

```ruby
def self.included(base)
  base.middleware.use ::Dynflow::Middleware::Common::Singleton
end
```

So that the people don't have to think about that (and forget) when they add it to their code. Also, a corresponding PR to docs would be nice while we have that in fresh memory.

Reply:
Good idea, will do

Reply:
so you can remove it now